docs/source/stability_policy.rst

```diff
@@ -0,0 +1,173 @@
+
+Introduction of new features or changes of flavors of comparable behavior (e.g. the default for the HTTP protocol version being used) are not covered by this policy.
+
+The internal structure of classes in PTB, i.e. things like the result of ``dir(obj))`` or the contents of ``obj.__dict__``, may change.
+
+Objects are in general not guaranteed to be pickleable (unless stated otherwise) and pickled objects from one version of PTB may not be loadable in future versions.
+We may provide a way to convert pickled objects from one version to another, but this is not guaranteed.
+
+Functionality that is part of PTBs API but is explicitly documented as not to be intended to be used directly by users (e.g. :meth:`telegram.request.BaseRequest.do_request`) may change.
+This also applies to functions or attributes marked as final in the sense of `PEP 591 <https://www.python.org/dev/peps/pep-0591/>`__.
+
+PTB has dependencies to 3rd party packages.
+The versions that PTB uses of these 3rd party packages may change if that does not change PTBs public API.
+
+PTB does not give guarantees about which Python versions are supported.
+In general, we will try to support all Python versions that have not yet reached their end of life, but we reserve ourselves the option to drop support for Python versions earlier if that benefits the advancement of the library.
+
+.. _bot-api-functionality-1:
+
+Bot API Functionality
+~~~~~~~~~~~~~~~~~~~~~
+
+Comparison of equality of instances of the classes in the :mod:`telegram` package is subject to change and the PTB team will update the behavior to best reflect updates in the Bot API.
+
+When the order of arguments of the Bot API methods changes or they become optional/mandatory due to changes in the Bot API, PTB will always try to reflect these changes.
+While we try to make such changes backward compatible, this is not always possible or only with significant effort.
+In such cases we will find a trade-off between backwards compatibility and fully complying with the Bot API, which may result in breaking changes.
+We highly recommend using keyword arguments, which can help to make such changes non-breaking on your end.
+
+..
+    We have documented a few common cases and possible backwards compatible solutions here in the wiki for reference for the dev team.
+
+To be moved to the wiki before merge:
+
+A few cases can happen when Telegram makes changes to arguments of classes and methods.
+Here are a few hints on how these can be handled in a compatible way that is backward compatible or at least minimizes the impact of the change.
+Whether these hints are applied or the change is just implemented in a breaking way may be decided by the dev team on a case by case basis.
+This depends e.g. on the currently available capacity.
+
+1. Optional arguments become mandatory
+
+   1. Keep the argument optional in the method signature
+   2. If the argument is not passed, raise a ``ValueError`` with a helpful message
```

Comment:
by the way, there's a difference here that I tried to follow when I was doing that `thumb_url` stuff.

If a required argument is not passed, Python raises `TypeError`, and the common practice AFAIK is to throw what Python throws. 

If both old and new are passed, I raise `ValueError`. 

That was my logic. I'm not agains raising `ValueError` in all cases for consistency, it's just that it's not consistent with default Python behavior.

Same for other cases below.

Reply:
No, you're completely right. I just didn't really think about that while tying …